### PR TITLE
PEER-232: Match Initiation UI

### PR DESCRIPTION
**Pull request by cyiting**
To test:
Go to: backend/matching/src/services/get-match-items.ts
Add a random roomId and 1<= questionId <= 20

e.g. {
    roomId: '12345',
    questionId: '4',
  };

### CODE REVIEW

File: frontend/src/routes/match/match-form.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                <FormItem>
+                  <FormLabel>Topics</FormLabel>
+                  <MultiSelector
+                    values={field.value}
+                    onValuesChange={field.onChange}
+                    className='whitespace-nowrap md:w-[350px] '
+                  >
+                    <FormControl>
+                      <MultiSelectorTrigger className='max-h-24 overflow-y-auto'>
+                        <MultiSelectorInput placeholder='Select topic(s)' />
```
Comment (cyiting) [inline frontend/src/routes/match/match-form.tsx]: TO DO: add radix ui caret sort icon if possible

File: frontend/.env.docker [hunk 2]
```diff
@@ -2,4 +2,5 @@ FRONTEND_ENV=local
 
 VITE_USER_SERVICE=http://host.docker.internal:9001
 VITE_QUESTION_SERVICE=http://host.docker.internal:9002
+VITE_COLLAB_SERVICE=http://host.docker.internal:9004
```
Comment (SeeuSim) [inline frontend/.env.docker]: add Matching Service

File: frontend/nginx.conf.template [hunk 3]
```diff
@@ -25,6 +25,15 @@ server {
         proxy_set_header X-Forwarded-Proto $scheme;
     }
 
+    location /matching-service/ {
+        rewrite ^/matching-service(/.*)$ $1 break;
+        proxy_pass ${VITE_MATCHING_SERVICE};
+        proxy_set_header Host $host;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Proto $scheme;
```
Comment (SeeuSim) [inline frontend/nginx.conf.template:39]: Has this been tested? i.e. ws proxy for socket.io

File: frontend/src/routes/match/match-form.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    setErrorMessage(null);
+
+    const response = await requestMatch(data);
+
+    if (!response) {
+      setLoading(false);
+      setErrorMessage('Error. Please try again later.');
+    } else {
+      const socketPort = response.socketPort;
+      navigate(ROUTES.WAITING_ROOM, { state: { socketPort } });
```
Comment (SeeuSim) [inline frontend/src/routes/match/match-form.tsx]: maybe just open a modal for D4 instead of navigating here

**Review by SeeuSim — CHANGES_REQUESTED** [review]
Lgtm, just test with integrated docker-compose config and see if nginx can proxy the ws connections

**Review by SeeuSim — APPROVED** [review]

**Pull request merged**